core/src/no/group15/playmagic/ecs/systems/BombExploderSystem.kt

```diff
@@ -0,0 +1,129 @@
+package no.group15.playmagic.ecs.systems
+
+import com.badlogic.ashley.core.*
+import com.badlogic.ashley.signals.Listener
+import com.badlogic.ashley.signals.Signal
+import com.badlogic.ashley.utils.ImmutableArray
+import com.badlogic.gdx.assets.AssetManager
+import com.badlogic.gdx.graphics.Texture
+import com.badlogic.gdx.graphics.g2d.TextureRegion
+import ktx.ashley.allOf
+import ktx.ashley.get
+import ktx.ashley.mapperFor
+import no.group15.playmagic.commands.Command
+import no.group15.playmagic.commands.CommandReceiver
+import no.group15.playmagic.commands.DropBombCommand
+import no.group15.playmagic.ecs.components.ExploderComponent
+import no.group15.playmagic.ecs.components.TextureComponent
+import no.group15.playmagic.ecs.components.TimerComponent
+import no.group15.playmagic.ecs.components.TransformComponent
+import no.group15.playmagic.ecs.entities.EntityFactory
+import no.group15.playmagic.events.BombTimeoutEvent
+import no.group15.playmagic.utils.assets.GameAssets
+
+
+class BombExploderSystem(
+	priority: Int,
+	private val assetManager: AssetManager
+): EntitySystem(
+	priority
+), Listener<BombTimeoutEvent>,
+	CommandReceiver {
+
+	private lateinit var entities: ImmutableArray<Entity>
+	private val timer = mapperFor<TimerComponent>()
+	private val texture = mapperFor<TextureComponent>()
+	private val transform = mapperFor<TransformComponent>()
+	private val exploder = mapperFor<ExploderComponent>()
+
+	override fun addedToEngine (engine: Engine) {
+		entities = engine.getEntitiesFor(
+			allOf(ExploderComponent::class, TimerComponent::class, TransformComponent::class, TextureComponent::class).get()
+		)
+		Command.Type.DROP_BOMB.receiver = this
+	}
+
+	/*
+	override fun update(deltaTime: Float) {
+		var explosionTexture: TextureRegion = TextureRegion(assetManager.get<Texture>(GameAssets.EXPLOSION.desc.fileName))
+
+		for (entity in entities) {
+			if (entity[timer]!!.timeLeft <= 0) {
+
+				entity[texture]!!.src = explosionTexture
+				entity[exploder]!!.isExploded = true
+
+			}
+		}
+	}
+	 */
+
+	override fun receive(signal: Signal<BombTimeoutEvent>, event: BombTimeoutEvent) {
+		// Exploded
+		if (event.bomb[exploder]!!.isExploded) {
+			engine.removeEntity(event.bomb)
+		}
+		//Not exploded
+		else {
+			event.bomb[texture]!!.src = TextureRegion(assetManager.get<Texture>(GameAssets.EXPLOSION.desc.fileName))
+			event.bomb[exploder]!!.isExploded = true
+
+			// create new timer
+			val newTimer = (engine as PooledEngine).createComponent(TimerComponent::class.java)
+			newTimer.timeLeft = 3f
+
+			event.bomb.add(newTimer)
+		}
+	}
+
+
+
+	override fun receive(command: Command) {
+		when (command) {
+			is DropBombCommand -> {
+				val bomb = EntityFactory.makeEntity(assetManager, engine as PooledEngine, EntityFactory.Type.BOMB)
+				bomb[timer]!!.timeLeft = 3f
+
+				// position
+				val playerPos = engine.getSystem(MovementSystem::class.java).localPlayerPosition()
```

Comment:
do a loop here with PlayerComponent family to find local player

Reply:
done!